### ISSUE TITLE
Trying to run container cluster on Mac OS

### ISSUE THREAD

**Issue by fastnsilver** [harbur/docker-rabbitmq-cluster#2]
My environment ...

Mac OS X El Capitan w/ Docker, Docker Machine and Docker Compose as follows:

```
________________________________________________________________________________
| ~ @ TheCaffeinator (cphi)
| => dm version
docker-machine version 0.7.0, build a650a40
________________________________________________________________________________
| ~ @ TheCaffeinator (cphi)
| => dc version
docker-compose version 1.7.1, build unknown
docker-py version: 1.8.1
CPython version: 2.7.10
OpenSSL version: OpenSSL 0.9.8zh 14 Jan 2016
________________________________________________________________________________
| ~ @ TheCaffeinator (cphi)
| => docker version
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.6.2
 Git commit:   5604cbe
 Built:        Wed Apr 27 15:27:26 UTC 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 19:36:04 2016
 OS/Arch:      linux/amd64
________________________________________________________________________________
| ~ @ TheCaffeinator (cphi)
| =>
```

When I try to `docker-compose up -d` w/ the `docker-compose.yml` file in your repo, I cannot start `rabbit1` instance.

Process output...

```
| => dc logs rabbit1
Attaching to docker_rabbit1_1
rabbit1_1   | Running as single server
________________________________________________________________________________
| ~/development/projects/opensource/xlator/deploy/docker @ TheCaffeinator (cphi)
| => dc ps
                Name                                   Command                                  State                                   Ports
-------------------------------------------------------------------------------------------------------------------------------------------------------------
docker_cadvisor_1                       /usr/bin/cadvisor                       Up                                      0.0.0.0:9080->8080/tcp
docker_data_1                           mkdir /data && chown 502:2 ...          Exit 1
docker_grafana_1                        /usr/sbin/grafana-server - ...          Up                                      0.0.0.0:3000->3000/tcp
docker_graphite_1                       /sbin/my_init                           Up                                      0.0.0.0:2003->2003/tcp,
                                                                                                                        0.0.0.0:8000->80/tcp,
                                                                                                                        0.0.0.0:8125->8125/tcp,
                                                                                                                        0.0.0.0:8125->8125/udp,
                                                                                                                        0.0.0.0:8126->8126/tcp
docker_rabbit1_1                        /pre-entrypoint.sh rabbitm ...          Exit 1
docker_rabbit2_1                        /pre-entrypoint.sh rabbitm ...          Up                                      0.0.0.0:15673->15672/tcp, 25672/tcp,
                                                                                                                        4369/tcp, 0.0.0.0:5673->5672/tcp,
                                                                                                                        9100/tcp, 9101/tcp, 9102/tcp,
                                                                                                                        9103/tcp, 9104/tcp, 9105/tcp
docker_rabbit3_1                        /pre-entrypoint.sh rabbitm ...          Up                                      15672/tcp, 25672/tcp, 4369/tcp,
                                                                                                                        0.0.0.0:5674->5672/tcp, 9100/tcp,
                                                                                                                        9101/tcp, 9102/tcp, 9103/tcp,
                                                                                                                        9104/tcp, 9105/tcp
```

Pay no attention to other containers I've started above.  

Anything I should be aware of? I cannot hit admin interface on port `15672` since `rabbit1` is down.

**Comment by dkapanidis** [comment]
Can you show the logs of rabbit1?

```
docker-compose logs rabbit1
```

Also try it with a fresh environment to make sure you don't have left-overs from previous runs:

```
docker-compose down
docker-compose up -d
```

And check if that fixes it. I tried locally and it works for me without issues.

**Comment by dkapanidis** [comment]
My environment: Mac Capitan, with Docker for Mac (1.11.1-beta11):

```
➜  docker-rabbitmq-cluster git:(master) docker --version
Docker version 1.11.1, build 5604cbe
➜  docker-rabbitmq-cluster git:(master) docker-compose --version
docker-compose version 1.7.1, build 0a9ab35
```

**Comment by fastnsilver** [comment]
I have a a couple of `.bash_profile` aliases for `docker-compose` and `docker-machine`, `dc` and `dm` respectively.  What follows is the sequence of steps I took on a new docker machine.  Still failing on startup.  I appear to be running the same docker versions as you.

```
| ~/development/projects/opensource @ TheCaffeinator (cphi)
| => git clone https://github.com/harbur/docker-rabbitmq-cluster.git
Cloning into 'docker-rabbitmq-cluster'...
remote: Counting objects: 29, done.
remote: Total 29 (delta 0), reused 0 (delta 0), pack-reused 29
Unpacking objects: 100% (29/29), done.
Checking connectivity... done.
________________________________________________________________________________
| ~/development/projects/opensource @ TheCaffeinator (cphi)
| => cd docker-rabbitmq-cluster/
total 28
drwxr-xr-x  12 cphi  staff   408B May 17 06:34 .git/
-rw-r--r--   1 cphi  staff   273B May 17 06:34 Dockerfile
-rw-r--r--   1 cphi  staff   1.1K May 17 06:34 LICENSE
-rw-r--r--   1 cphi  staff   1.4K May 17 06:34 README.md
-rw-r--r--   1 cphi  staff    71B May 17 06:34 captain.yml
-rw-r--r--   1 cphi  staff   572B May 17 06:34 docker-compose.yml
-rwxr-xr-x   1 cphi  staff   594B May 17 06:34 pre-entrypoint.sh
-rwxr-xr-x   1 cphi  staff   532B May 17 06:34 rabbitmq-cluster
________________________________________________________________________________
| ~/development/projects/opensource/docker-rabbitmq-cluster @ TheCaffeinator (cphi)
| => dc up -d
Creating dockerrabbitmqcluster_rabbit1_1
Creating dockerrabbitmqcluster_rabbit2_1

ERROR: for rabbit2  Cannot link to a non running container: /dockerrabbitmqcluster_rabbit1_1 AS /dockerrabbitmqcluster_rabbit2_1/dockerrabbitmqcluster_rabbit1_1
Traceback (most recent call last):
  File "/usr/local/Cellar/docker-compose/1.7.1/libexec/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.7.1', 'console_scripts', 'docker-compose')()
  File "/usr/local/Cellar/docker-compose/1.7.1/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 63, in main
    log.error(e.msg)
AttributeError: 'ProjectError' object has no attribute 'msg'
________________________________________________________________________________
| ~/development/projects/opensource/docker-rabbitmq-cluster @ TheCaffeinator (cphi)
| => dc logs rabbit1
Attaching to dockerrabbitmqcluster_rabbit1_1
rabbit1_1  | Running as single server
| ~/development/projects/opensource/docker-rabbitmq-cluster @ TheCaffeinator (cphi)
| => dc ps
             Name                            Command                State     Ports
-----------------------------------------------------------------------------------
dockerrabbitmqcluster_rabbit1_1   /pre-entrypoint.sh rabbitm ...   Exit 1
dockerrabbitmqcluster_rabbit2_1   /pre-entrypoint.sh rabbitm ...   Exit 128
```

**Comment by fastnsilver** [comment]
This looks like a bug, here: https://github.com/docker/compose/issues/3432, to be fixed in docker 1.8.0?

**Comment by fastnsilver** [comment]
Turned out I had exceeded the disk space on my docker-machine.  I provisioned another machine, and everything worked beautifully!  Apologies.  Thanks!
